.github/workflows/maven.yml

```diff
@@ -0,0 +1,44 @@
+name: Axon Hotel Demo
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - master
+
+jobs:
+  build:
+    name: Test and Build on JDK ${{ matrix.java-version }}
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        include:
+          - java-version: 11
+          - java-version: 17
+      fail-fast: false
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK ${{ matrix.java-version }}
+        uses: actions/setup-java@v2.5.0
+        with:
+          distribution: 'zulu'
+          java-version: ${{ matrix.java-version }}
+          server-id: sonatype
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+
+      - name: Cache .m2
```

Comment:
`setup-java` already has [cache](https://github.com/actions/setup-java#caching-maven-dependencies) built-in!
you can just add a `cache: maven` to their options

Reply:
Ah great, I'll edit that and adjust the Gift Card demo as well (I stole it from there)